uaclient/tests/test_reboot_cmds.py

```diff
@@ -122,3 +128,40 @@ def test_run_command_failure(self, m_subp, m_exit, caplog_text):
             mock.call("marker-reboot-cmds")
         ]
         assert m_exit.call_args_list == [mock.call(1)]
+
+
+class TestProcessRebootOperations:
+    @pytest.mark.parametrize("caplog_text", [logging.DEBUG], indirect=True)
+    @mock.patch("uaclient.config.UAConfig.delete_cache_key")
+    @mock.patch("uaclient.config.UAConfig.check_lock_info")
+    @mock.patch("uaclient.config.UAConfig.add_notice")
+    @mock.patch("lib.reboot_cmds.fix_pro_pkg_holds")
+    def test_process_reboot_operations_create_notice_when_it_fails(
+        self,
+        m_fix_pro_pkg_holds,
+        m_add_notice,
+        m_check_lock_info,
+        _m_delete_cache_key,
+        FakeConfig,
+        caplog_text,
+    ):
+        m_check_lock_info.return_value = (0, 0)
+        m_fix_pro_pkg_holds.side_effect = ProcessExecutionError("error")
+        args = mock.MagicMock()
+
+        cfg = FakeConfig()
+        cfg.for_attached_machine()
+        with mock.patch("os.path.exists", return_value=True):
+            with mock.patch("uaclient.config.UAConfig.write_cache"):
+                process_reboot_operations(args, cfg)
+
+        assert (
+            mock.call("", MESSAGE_REBOOT_SCRIPT_FAILED)
+            in m_add_notice.call_args_list
+        )
```

Comment:
Based on the issue description, I would also expect an assertion similar to this one but checking that `remove_notice` was called for the "Operation in progress..." notice. Or perhaps a check that the lock was released properly.